core/src/test/java/org/elasticsearch/action/search/SearchResponseTests.java

```diff
@@ -0,0 +1,120 @@
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentParserUtils;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.search.aggregations.InternalAggregations;
+import org.elasticsearch.search.internal.InternalSearchHit;
+import org.elasticsearch.search.internal.InternalSearchHits;
+import org.elasticsearch.search.internal.InternalSearchHitsTests;
+import org.elasticsearch.search.profile.SearchProfileShardResults;
+import org.elasticsearch.search.suggest.Suggest;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertToXContentEquivalent;
+
+public class SearchResponseTests extends ESTestCase {
+
+    public static SearchResponse createTestItem() {
+        InternalSearchHits hits = InternalSearchHitsTests.createTestItem();
+        boolean timedOut = randomBoolean();
+        Boolean terminatedEarly = randomBoolean() ? null : randomBoolean();
+        int tookInMillis = randomIntBetween(0, 1000);
+        int successfulShards = tookInMillis;
+        int totalShards = randomInt();
+        int numFailures = randomIntBetween(0, 3);
+        ShardSearchFailure[] failures = new ShardSearchFailure[numFailures];
+        for (int i = 0; i < numFailures; i++) {
+            failures[i] = ShardSearchFailureTests.createTestItem();
+        }
+        // TODO add random aggregations, suggest and profileShardResults once we are able to parse them from xContent
+        InternalAggregations aggregations = null;
+        Suggest suggest = null;
+        SearchProfileShardResults profileShardResults = null;
+        return new SearchResponse(new InternalSearchResponse(hits, aggregations, suggest, profileShardResults, timedOut, terminatedEarly),
+                null, totalShards, successfulShards, tookInMillis, failures);
+    }
+
+    public void testFromXContent() throws IOException {
+        SearchResponse response = createTestItem();
+        XContentType xcontentType = randomFrom(XContentType.values());
+        XContentBuilder builder = XContentFactory.contentBuilder(xcontentType);
+        builder = response.toXContent(builder, ToXContent.EMPTY_PARAMS);
+
+        XContentParser parser = createParser(builder);
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+        SearchResponse parsed = SearchResponse.fromXContent(parser);
+
+        // the "_shard/total/failures" section makes if impossible to directly compare xContent, because
+        // the failures in the parsed SearchResponse are wrapped in an extra ElasticSearchException on the client side.
+        // Because of this we compare the "top level" fields for equality and the subsections xContent equivalence independently
```

Comment:
Good point, I will change this. However, I'd like the createTestItem() method to still be able to output an object with set "failures", so I will make this controllable from the outside. In our "normal" fromXContent() test we can use xContent comparison again then, In another test with "failures" at the moment the only thing we can really check is that we can parse the response without throwing an error I think. Other more detailed tests are covered by ShardSearchFailureTests already I think.